src/interpreter/post_execution.rs

```diff
@@ -18,19 +20,23 @@ where
     ///
     /// The transaction validation is expected to halt in such case. Since the VM only accepts
     /// checked transactions - hence, validated - this case should be unreachable.
-    pub(crate) fn finalize_outputs(&mut self, revert: bool) -> Result<(), RuntimeError> {
-        let outputs = self.tx.transaction().outputs().len();
-        let params = &self.params;
-        let tx = &mut self.tx;
-
-        let remaining_gas = self.registers[REG_GGAS];
-
-        tx.update_outputs(params, revert, remaining_gas, &self.balances)
+    pub(crate) fn finalize_outputs<Tx>(
```

Comment:
If we don't need this method for `Create`(TODO: link on the comment above), then I will revert this change, and it will be used only for `Script` =)